lib/matplotlib/colors.py

```diff
@@ -876,10 +876,42 @@ def __init__(self, vmin=None, vmax=None, clip=False):
         the over, under, and masked colors in the colormap, so it is
         likely to lead to surprises; therefore the default is
         *clip* = *False*.
+
+        You can pass an array-like object to *data*, e.g. a list of
+        numpy arrays. If *vmin* or *vmax* are *None*, the respective min/max
+        of the array is taken as *vmin* / *vmax*.
+
         """
         self.vmin = _sanitize_extrema(vmin)
         self.vmax = _sanitize_extrema(vmax)
         self.clip = clip
+        if data is not None:
+            self.autoscale_None(data)
```

Comment:
Did you mean `self.apply_autoscale` here or am I misunderstanding?  I'm not seeing how data can be a list of np arrays if you call `autoscale_None`.  I don't see that you have a test of this below, i.e. you only call `apply_autoscale`.

Reply:
No. I mean `autoscale_None`. `autoscale_None` internally calls `np.asanyarray()` on the data, so a list of arrays is fine. I don't think we really need to test this, but if you want I can add a test.

`norm = Normalize(data=data)` is really just a shortcut for
~~~
norm = Normalize()
norm.autoscale(data)
~~~
i.e. it scales to the data. The `autoscale_None` variant is used so that you  can consistently do `Normalize(vmin=0, data=data)` in which case only vmax will be taken from data.

`apply_autoscale()` serves a different purpose. While the above works on data, `apply_autoscale()` works on `AxesImages`. It creates a norm based on the data in the images and applies the norm to the images. Maybe we should better call it `autoscale_images()`?